### PR TITLE
fix: more expressive errors

### LINKED ISSUE #22446: Disk not supporting hard links creates weird error from cli during backup
Bad error message:

```
Error: failed to backup bucket data: failed to download snapshot of shard 1: download 1: InfluxDB OSS-only command failed: 500 Internal Server Error: An internal error has occurred
```

### PR DESCRIPTION
**Pull request by lesam**
Closes #22446

Instead of "internal server error", tell the client that the error is in the server logs, and actually put them there.

- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [ ] Tests pass
- [ ] http/swagger.yml updated (if modified Go structs or API)
- [ ] Feature flagged (if modified API)

### CODE REVIEW

File: kit/transport/http/error_handler.go [hunk 1]
```diff
@@ -28,11 +28,7 @@ func (h ErrorHandler) HandleHTTPError(ctx context.Context, err error, w http.Res
 
 	code := errors2.ErrorCode(err)
 	var msg string
-	if err, ok := err.(*errors2.Error); ok {
-		msg = err.Error()
-	} else {
-		msg = "An internal error has occurred"
-	}
+	msg = err.Error()
```
Comment (danxmoran) [inline kit/transport/http/error_handler.go:31]: Do we want to keep `An internal error has occurred` as a prefix for the returned message? Or is that just noise?

Comment (lesam) [reply kit/transport/http/error_handler.go:31]: Unless we want to tag it as an error not in the expected format.

Comment (lesam) [reply kit/transport/http/error_handler.go:31]: Something like `unknown error type:`

**Comment by lesam** [comment]
Fixed linter complaint

File: kit/transport/http/error_handler.go [hunk 2]
```diff
@@ -28,10 +35,11 @@ func (h ErrorHandler) HandleHTTPError(ctx context.Context, err error, w http.Res
 
 	code := errors2.ErrorCode(err)
 	var msg string
-	if err, ok := err.(*errors2.Error); ok {
+	if _, ok := err.(*errors2.Error); ok {
 		msg = err.Error()
 	} else {
-		msg = "An internal error has occurred"
+		msg = "An internal error has occurred - check server logs"
+		h.logger.Warn("internal error not returned to client", zap.Error(err))
```
Comment (lesam) [inline kit/transport/http/error_handler.go:42]: @dgnorton suggested (and I agree) it is probably too insecure to return any and every error back to the client. Instead, log the error and tell the client to check the server logs - anyone with server log access should have permission to see whatever errors we create.

Comment (lesam) [reply kit/transport/http/error_handler.go:42]: Particular errors that we actually want to return to the client should be converted to the correct format to be returned.

Comment (lesam) [reply kit/transport/http/error_handler.go:42]: New error from client:

```
Error: failed to backup bucket data: failed to download snapshot of shard 1: InfluxDB OSS-only command failed: 500 Internal Server Error: An internal error has occurred - check server logs
```

From server logs:

```
2021-09-13T15:45:36.828559Z     warn    internal error not returned to client   {"log_id": "0WZyCYXW000", "handler": "error_logger", "error": "error creating tsm hard link: hard coded failure for /Users/sarnold/.influxdbv2/engine/data/b029e667ccf6262d/autogen/1/000000015-000000002.tsm"}
```

**Review by danxmoran — APPROVED** [review]

**Pull request merged**